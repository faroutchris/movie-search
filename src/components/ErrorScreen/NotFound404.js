import React from "react";
import { Link } from "react-navi";
const NotFound = () => (
  <div>
    <h1>404 Not found</h1>
    <Link href="/">Back to start page</Link>
  </div>
);

export default NotFound;
