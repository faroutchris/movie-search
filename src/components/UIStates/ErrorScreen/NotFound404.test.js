import React from "react";
import ReactDOM from "react-dom";
import TestRenderer from "react-test-renderer";
import NotFound404 from "./NotFound404";

describe("NotFound404", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");

    ReactDOM.render(<NotFound404 />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test("NotFound404 snapshot", () => {
    const component = TestRenderer.create(<NotFound404 />);

    const tree = component.toJSON();
    const link = tree.children.find(child => child.type === "a");

    expect(link.props.href).toBe("/");

    expect(tree).toMatchSnapshot();
  });
});
