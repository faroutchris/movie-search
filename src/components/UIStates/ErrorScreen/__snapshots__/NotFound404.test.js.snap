// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound404 NotFound404 snapshot 1`] = `
<div>
  <h1>
    404 Not found
  </h1>
  <a
    className=" "
    context={Object {}}
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    Back to start page
  </a>
  <br />
  <br />
</div>
`;
